### PR TITLE
chore: added unmarshal and validate capability

### PR DESCRIPTION
**Pull request by ricardomaraschini**
added an auxiliar function to allow for linting directly out of a yaml
file content. this function parses the body into an intermediary struct
and then applies regular linting rules on top of the resulting object.

this commit also adds a considerable amount of changes in the linting
process, these changes were introduced after using the lint on top of a
database of 99k installer objects and capturing some problems with the
linting logic.

### CODE REVIEW

File: pkg/lint/tests/rego/openebs_gte_2.12.9_cstor.yaml [hunk 1]
```diff
@@ -10,6 +10,6 @@ installer:
       version: 3.3.0
       isCstorEnabled: true
 output: 
-  - message: OpenEBS version 3.3.0 does not support cStor in kurl
+  - message: OpenEBS versions >= 2.12.9 dont support cStor in kurl
```
Comment (emosbaugh) [inline pkg/lint/tests/rego/openebs_gte_2.12.9_cstor.yaml:13]:
```suggestion
  - message: OpenEBS add-on versions >= 2.12.9 do not support cStor
```

File: pkg/lint/rego/output.rego [hunk 2]
```diff
@@ -194,13 +226,11 @@ lint[output] {
...
-	input.spec.openebs.isCstorEnabled
+	installer.spec.openebs.isCstorEnabled
 	is_addon_version_greater_than_or_equal("openebs", "2.12.9")
-	version := input.spec.openebs.version
-	message := sprintf("OpenEBS version %v does not support cStor in kurl", [version])
 	output := {
-		"type": "misconfiguration",
-		"message": message,
+		"type": "incompatibility",
+		"message": "OpenEBS versions >= 2.12.9 dont support cStor in kurl",
```
Comment (emosbaugh) [inline pkg/lint/rego/output.rego:233]:
```suggestion
		"message": "OpenEBS add-on versions >= 2.12.9 do not support cStor",
```

File: pkg/lint/tests/rego/pod_cidr_only_mask.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      version: 2.6.0
+    minio:
+      version: 2020-01-25T02-50-51Z
+    contour:
+      version: 1.18.0
+    registry:
+      version: 2.7.1
+    kotsadm:
+      version: 1.56.0
+      applicationSlug: puppet-application-manager
```
Comment (emosbaugh) [inline pkg/lint/tests/rego/pod_cidr_only_mask.yaml:18]: lets remove our vendor's names from the project

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: pkg/lint/tests/rego/pod_cidr_only_mask.yaml (hunk 3 above)
Comment (ricardomaraschini) [reply pkg/lint/tests/rego/pod_cidr_only_mask.yaml]: Fixed.

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**